src/lib/one-graph/client.js

```diff
@@ -0,0 +1,835 @@
+    nfToken: authToken,
+    appId,
+    id: docId,
+  })
+
+  const persistedQuery = response.data && response.data.oneGraph && response.data.oneGraph.persistedQuery
+
+  return persistedQuery
+}
+
+const fetchCliSessionEvents = async ({ appId, authToken, sessionId }) => {
+  const first = 1000
+  const next = await fetchOneGraph(null, appId, internalOperationsDoc, 'CLISessionEventsQuery', {
+    nfToken: authToken,
+    sessionId,
+    first,
+  })
+
+  if (next.errors) {
+    return next
+  }
+
+  const events = (next.data && next.data.oneGraph && next.data.oneGraph.netlifyCliEvents) || []
+
+  return { events }
+}
+
+const monitorCLISessionEvents = ({
+  appId,
+  authToken,
+  netligraphConfig,
+  onClose,
+  onError,
+  onEvents,
+  sessionId,
+  state,
+}) => {
+  const frequency = 5000
+  let shouldClose = false
+
+  const enabledServiceWatcher = async (netlifyToken, siteId) => {
+    const enabledServices = state.get('oneGraphEnabledServices') || ['onegraph']
+    const enabledServicesInfo = await fetchEnabledServices(netlifyToken, siteId)
+    if (!enabledServicesInfo) {
+      console.warn('Unable to fetch enabled services for site for code generation')
+      return
+    }
+    const newEnabledServices = enabledServicesInfo.map((service) => service.service)
+    const enabledServicesCompareKey = enabledServices.sort().join(',')
+    const newEnabledServicesCompareKey = newEnabledServices.sort().join(',')
+
+    if (enabledServicesCompareKey !== newEnabledServicesCompareKey) {
+      log(
+        `${NETLIFYDEVLOG} ${chalk.magenta(
+          'Reloading',
+        )} Netlify Graph schema..., ${enabledServicesCompareKey} => ${newEnabledServicesCompareKey}`,
+      )
+      await refetchAndGenerateFromOneGraph({ netligraphConfig, state, netlifyToken, siteId })
+      log(`${NETLIFYDEVLOG} ${chalk.green('Reloaded')} Netlify Graph schema and regenerated functions`)
+    }
+  }
+
+  const close = () => {
+    shouldClose = true
+  }
+
+  let handle
+
+  const helper = async () => {
+    if (shouldClose) {
+      clearTimeout(handle)
+      onClose()
+    }
+
+    const next = await fetchCliSessionEvents({ appId, authToken, sessionId })
+
+    if (next.errors) {
+      next.errors.forEach((error) => {
+        onError(error)
+      })
+    }
+
+    const { events } = next
+
+    if (events.length !== 0) {
+      const ackIds = onEvents(events)
+      await ackCLISessionEvents({ appId, authToken, sessionId, eventIds: ackIds })
+    }
+
+    await enabledServiceWatcher(authToken, appId)
+
+    handle = setTimeout(helper, frequency)
+  }
+
+  // Fire immediately to start rather than waiting the initial `frequency`
+  helper()
+
+  return close
+}
+
+const createCLISession = async (netlifyToken, appId, name, metadata) => {
+  const payload = {
+    nfToken: netlifyToken,
+    appId,
+    name,
+    metadata,
+  }
+
+  const result = await fetchOneGraph(null, appId, internalOperationsDoc, 'CreateCLISessionMutation', payload)
+
+  const session =
```

Comment:
@erezrokah, @ehmicky when can we drop node 12 support? 

This is a perfect example for optional chaining would enhance readability so much. 
😢